portal/app/algorithms/battery_load_control.py

```diff
@@ -0,0 +1,52 @@
+import json, datetime, numpy as np
+from app.farm_updater import sonnen_api
+from django.conf import settings
+
+
+def batt_dispatch():
+    from app.models import FarmDevice, FarmData, DeviceType
+
+    try:
+        egauge_device = FarmDevice.objects.get(device_uid=settings.EGAUGE_ID)
+    except Exception as exc:
+        print('Error: ', exc)
+
+    farmdata = FarmData.objects.filter(farm_device=egauge_device).order_by('-id')[0:3]
+    batt_instance = sonnen_api.SonnenApiInterface()
+    test_pen_power = []
+
+    # Getting egauge info
+    for data in farmdata:
+        egauge_data = json.loads(data.device_data)
+        test_pen_power.append(egauge_data['processed']['POWER_CIRCUIT1'] + egauge_data['processed']['POWER_CIRCUIT2'])
+
+    avg_test_pen_power = np.average(np.asarray(test_pen_power))
+
+    hour_day = datetime.datetime.now().hour - 7
+
+    # getting batt soc
+    sonnen_queryset = FarmDevice.objects.filter(type=DeviceType.SONNEN)
+    for query in sonnen_queryset:
```

Comment:
A QuerySet contains the models that were queried for. The name `query` here is very confusing. Name it `item`, `entity`, `model`, `element` or even just `farm_device`. Any of these would be acceptable.